enclave/src/lib.rs

```diff
@@ -107,6 +109,9 @@ pub enum Timeout {
 pub type Hash = sp_core::H256;
 type BPool = BasicPool<SideChainApi<Block>, Block>;
 
+use crate::error::Error;
+pub type EnclaveResult<T> = Result<T, Error>;
```

Comment:
is there a specific reason you're implementing this here and not in the error.rs file?
(like here: https://github.com/integritee-network/worker/blob/master/enclave/src/top_pool/error.rs#L27)

Reply:
My impression is that the error.rs should only contain the `Error` and not also the result - just in terms of semantic grouping.

Reply:
In terms of usability as well? If I use the error type, I most likely will use the result as well.

Atlest substrate is putting the result wihtin error.rs (if they have implemented one, which is more often not the case than it is): 
https://github.com/paritytech/substrate/blob/master/primitives/transaction-pool/src/error.rs#L25
https://github.com/paritytech/substrate/blob/master/primitives/blockchain/src/error.rs#L28

Reply:
But I'll leave this decision to the more expirienced ones - just let me know what the way to go is.

Reply:
No it's fine, I will change that. I wasn't sure myself. 

In contrast to your substrate examples, I had the impression that it's usually not in the error file, but it seems I was wrong.